rfc/rfc-0005-customizable-consumption-offset.md

```diff
@@ -0,0 +1,49 @@
+# RFC-#: Title
+
+**Authors:** Eric Bottard
+
+**Status:**
+
+**Pull Request URL:** link
+
+**Superseded by:** N/A
+
+**Supersedes:** N/A
+
+**Related:** rfc-0003: developer utils
+
+
+## Problem
+Sometimes users want to replay data through functions, maybe because the function logic has changed, or maybe because downstream systems have lost state and that state needs to be re-created. The current Processor concept in riff doesn't support such a mechanism though, and always consumes data from the (current) end of streams.
+
+### Anti-Goals
+
+## Solution
+With a top to bottom view of the user experience, the proposed solution is to
+- introduce an offset option on each input binding when a processor is created
+- persist that option in the Processor CRD instance on the cluster
+- have the Processor reconciler forward that option to the streaming-processor, so that it changes its SubscriptionRequests accordingly
+
+### User Impact
+This RFC proposes to change the `riff streaming processor create` command from
```

Comment:
We should consider the impact on the CRDs first, since that is the public API for riff.

Reply:
Added section about CRD changes